storage/store.go

```diff
@@ -1030,6 +1038,14 @@ func (s *Store) RemoveRange(rng *Range) error {
 	return nil
 }
 
+// PostProcessAfterSnapshot post-processes a range after a Raft
+// snapshot is applied and the range's descriptor has changed. It
+// currently just adds the range to the rangesByKey slice.
+func (s *Store) PostProcessAfterSnapshot(rng *Range) {
+	s.rangesByKey = append(s.rangesByKey, rng)
```

Comment:
Thanks for the review, @bdarnell .

I see. Added `uninitRanges` to `Store` to keep track of uninitialized ranges. It is used to avoid adding the same range more than once to `rangesByKey`. `uninitRanges` could be also used to garbage collect orphaned uninitialized ranges, I guess.

Sure, I've renamed the method to `ProcessRangeDescriptorUpdate` and changed the code to call it from `AdminSplit`, etc. I wondered `splitTrigger` might be a better place, but I wasn't totally sure.

Yes, it needs a lock... Fixed.

I've added a small check to `addRangeInternal` and made the method return an error if a given range hasn't yet been initialized. Could you take another look?